cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -181,37 +181,41 @@ import           System.FilePath
 -- sense under some 'ElaboratedSharedConfig'.
 sanityCheckElaboratedConfiguredPackage :: ElaboratedSharedConfig
                                        -> ElaboratedConfiguredPackage
-                                       -> Bool
+                                       -> a
+                                       -> a
 sanityCheckElaboratedConfiguredPackage sharedConfig
-                                       pkg@ElaboratedConfiguredPackage{..} =
+                                       pkg@ElaboratedConfiguredPackage{..}
+                                       ret =
 
     -- we should only have enabled stanzas that actually can be built
     -- (according to the solver)
-    pkgStanzasEnabled `Set.isSubsetOf` pkgStanzasAvailable
+    assert (pkgStanzasEnabled `Set.isSubsetOf` pkgStanzasAvailable)
```

Comment:
Yes. I just converted it into multiple asserts so that we know WHICH one failed when one fails. But arguably we should have a proper assert mechanism that is decoupled from optimization.